src/managers/controller.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling KRaft Controller."""
+
+import logging
+import os
+from subprocess import CalledProcessError
+from typing import TYPE_CHECKING
+
+from tenacity import retry, stop_after_attempt, wait_fixed
+
+from core.cluster import ClusterState
+from core.workload import WorkloadBase
+from literals import GROUP, KRAFT_VERSION, USER
+
+if TYPE_CHECKING:
```

Comment:
**question** what's is this for?

Reply:
Oops, missed this one :) Thanks for eying that!